### ISSUE TITLE
Can Android also be decompiled

### ISSUE THREAD

**Issue by fgknnc** [char-ptr/gdke#21]
Can apk also be decompiled

**Comment by PaologGithub** [comment]
> Can apk also be decompiled

Nope. GDKE uses DLL injection, which is only for Windows. I don't know if macOS/Linux supports it, but it can't be for mobile (iOS, iPadOS, Android)

**Comment by fgknnc** [comment]
? > > No. GDKE uses DLL injection and is only available for Windows. I don't know if macOS/Linux supports it, But it's not available for mobile devices (iOS, iPad OS, Android).

 How do I find the encryption key for the gde file on Android?

**Comment by PaologGithub** [comment]
> ? > > No. GDKE uses DLL injection and is only available for Windows. I don't know if macOS/Linux supports it, But it's not available for mobile devices (iOS, iPad OS, Android).
> 
> How do I find the encryption key for the gde file on Android?

I don't really know. You can't really inject things into an apk. Try using apktool, and watch the files to see if somewhere there is a trace of the key

**Comment by char-ptr** [comment]
your best bet would be to use ida/ghidra/etc to reverse the binary and find the encryption key. i have a guide in the description although might be outdated for newer godot games

**Comment by fgknnc** [comment]
How can I obtain the encryption key for an Android game made with Godot?

**Comment by char-ptr** [comment]
> How can I obtain the encryption key for an Android game made with Godot?

read my previous message c:

**Comment by fgknnc** [comment]
> > How do I get the encryption key for an Android game made with Godot? > > Read my previous message C: I inverted the apk binary with ida, but I didn't find the gdscript::load_byte_code function

**Comment by fgknnc** [comment]
> > How can I obtain the encryption key for an Android game made with Godot?
> 
> read my previous message c:

 I inverted the apk binary with ida, but I didn't find the gdscript::load_byte_code function

**Comment by PaologGithub** [comment]
> > > How can I obtain the encryption key for an Android game made with Godot?
> > 
> > 
> > read my previous message c:
> 
> I inverted the apk binary with ida, but I didn't find the gdscript::load_byte_code function

One new godot (4.x), it isn't anymore this function. I don't know what this is now, but the gdscript::load_byte_code is for godot 3.x

**Comment by fgknnc** [comment]
> > > > > > > > > Read my previous message C: > > > > > > I reversed the apk binary with ida, but didn't find the gdscript: load _ byte _ code function > > a new godot (4.x), no longer this function. I don't know what this is now, but gdscript: load _ byte _ code works for godot 3.x

 它是godot3.2制作的

**Comment by fgknnc** [comment]
> > > > How can I obtain the encryption key for an Android game made with Godot?
> > > 
> > > 
> > > read my previous message c:
> > 
> > 
> > I inverted the apk binary with ida, but I didn't find the gdscript::load_byte_code function
> 
> One new godot (4.x), it isn't anymore this function. I don't know what this is now, but the gdscript::load_byte_code is for godot 3.x

It is made by Godot 3.2

**Comment by char-ptr** [comment]
> > > > How can I obtain the encryption key for an Android game made with Godot?
> > > 
> > > 
> > > read my previous message c:
> > 
> > 
> > I inverted the apk binary with ida, but I didn't find the gdscript::load_byte_code function
> 
> One new godot (4.x), it isn't anymore this function. I don't know what this is now, but the gdscript::load_byte_code is for godot 3.x

it's now this function https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/core/io/file_access_pack.cpp#L150
(specifically this part https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/core/io/file_access_pack.cpp#L246-L260)

**Comment by fgknnc** [comment]
> > > > > How can I obtain the encryption key for an Android game made with Godot?
> > > > 
> > > > 
> > > > read my previous message c:
> > > 
> > > 
> > > I inverted the apk binary with ida, but I didn't find the gdscript::load_byte_code function
> > 
> > 
> > One new godot (4.x), it isn't anymore this function. I don't know what this is now, but the gdscript::load_byte_code is for godot 3.x
> 
> it's now this function https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/core/io/file_access_pack.cpp#L150 (specifically this part https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/core/io/file_access_pack.cpp#L246-L260)

I found open_and_marse, but it's not a function. What should I do next

**Comment by fgknnc** [comment]
> > > > > How can I obtain the encryption key for an Android game made with Godot?
> > > > 
> > > > 
> > > > read my previous message c:
> > > 
> > > 
> > > I inverted the apk binary with ida, but I didn't find the gdscript::load_byte_code function
> > 
> > 
> > One new godot (4.x), it isn't anymore this function. I don't know what this is now, but the gdscript::load_byte_code is for godot 3.x
> 
> it's now this function https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/core/io/file_access_pack.cpp#L150 (specifically this part https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/core/io/file_access_pack.cpp#L246-L260)

I found open_and_parse, but it's not a function. What should I do next
